.github/workflows/spelling2.yml

```diff
@@ -57,35 +81,57 @@ jobs:
       contents: read
       pull-requests: read
       actions: read
+      security-events: write
```

Comment:
this enables check-spelling to publish sarif reports. It works for Pushes and PRs w/in a repository, but not for forks.

these reports can be pulled down into tools like VS Code and provide ~5000 entries (whereas there's a limit of closer to 10 items per category being shown in GitHub annotations).